packages/near-fast-auth-signer/src/lib/useAuthState.ts

```diff
@@ -10,90 +11,95 @@ import { redirectWithError, safeGetLocalStorage } from '../utils';
   const [query] = useSearchParams();
   const email = query.get('email');
   const successUrl = query.get('success_url');
   const failureUrl = query.get('failure_url');
 
-  if (webauthnUsername === undefined) {
-    return { authenticated: new Error('Please allow third party cookies') };
-  }
-
   useEffect(() => {
     const handleAuthState = async () => {
-      const controllerState = await window.fastAuthController.isSignedIn();
-      const isPasskeySupported = await isPassKeyAvailable();
-
-      if (skipGetKeys) {
-        setAuthenticated(false);
-      } else if (controllerState === true) {
-        setAuthenticated(true);
-      } else if (isPasskeySupported && (!webauthnUsername || (email && email !== webauthnUsername))) {
-        setAuthenticated(false);
-      } else if (isPasskeySupported) {
-        try {
-          const keypairs = await getKeys(webauthnUsername);
-          const accounts = await Promise.allSettled(
-            keypairs.map(async (k) => {
-              const accIds = await fetchAccountIds(k.getPublicKey().toString());
-              return accIds.map((accId) => { return { accId, keyPair: k }; });
-            })
-          );
-
-          const accountsList = accounts
-            .flatMap((result) => (result.status === 'fulfilled' ? result.value : []));
-
-          if (accountsList.length === 0) {
-            setAuthenticated(false);
-          }
-
-          window.fastAuthController = new FastAuthController({
-            accountId: accountsList[0].accId,
-            networkId
-          });
-
-          await window.fastAuthController.setKey(new KeyPairEd25519(accountsList[0].keyPair.toString().split(':')[1]));
-          setAuthenticated(true);
-        } catch {
-          setAuthenticated(false);
+      try {
+        if (!email) {
+          throw new Error('Email is required');
         }
-      } else {
-        checkFirestoreReady().then(async (isReady) => {
-          if (isReady) {
-            const oidcToken = await firebaseAuth.currentUser.getIdToken();
-            const localStoreKey = await window.fastAuthController.getLocalStoreKey(`oidc_keypair_${oidcToken}`);
-            if (localStoreKey) {
-              const recoveryPK = await window.fastAuthController.getUserCredential(oidcToken);
-              const accountIds = await fetchAccountIds(recoveryPK);
-              (window as any).fastAuthController = new FastAuthController({
-                accountId: accountIds[0],
-                networkId
-              });
-              setAuthenticated(true);
-            } else {
-              setAuthenticated(false);
-            }
-          }
-        });
-      }
-    };
 
-    handleAuthState()
-      .catch((e) => {
+        setAuthenticated(await getAuthState(email, skipGetKeys));
+      } catch (e) {
+        captureException(e);
         redirectWithError({
           failure_url: failureUrl,
           success_url: successUrl,
           error:       e.message,
         });
```

Comment:
I know this was introduced in a separate PR but do we want to handle redirects from this hook? It strikes me as an unnecessary side effect that should be handled in the specific cases that require it outside this hook.